client/ayon_deadline/plugins/publish/blender/submit_blender_deadline.py

```diff
@@ -24,7 +24,70 @@ class BlenderSubmitDeadline(abstract_submit_deadline.AbstractSubmitDeadline,
     families = ["render"]  # TODO this should be farm specific as render.farm
     settings_category = "deadline"
 
-    def get_job_info(self, job_info=None):
+    def process(self, instance):
+        if not instance.data.get("farm"):
+            self.log.debug("Render on farm is disabled. "
+                           "Skipping deadline submission.")
+            return
+
+        # Always set instance output directory to the expected
+        expected_files = instance.data["expectedFiles"]
+        if not expected_files:
+            raise PublishError(
+                message="No Render Elements found.",
+                title="No Render Elements found.",
+                description="Expected files for render elements are empty."
+            )
+
+        first_file = next(iter_expected_files(expected_files))
+        output_dir = os.path.dirname(first_file)
+        instance.data["outputDir"] = output_dir
+        instance.data["toBeRenderedOn"] = "deadline"
+
+        # We are submitting a farm job not per instance - but once per Blender
+        # scene. This is a hack to avoid submitting multiple jobs for each
+        # comp file output because the Deadline job will always render all
+        # active ones anyway (and the relevant view layers).
+        context = instance.context
+        key = "__hasRun{}".format(self.__class__.__name__)
+        if context.data.get(key, False):
+            return
+        else:
+            context.data[key] = True
+
+        # Collect all saver instances in context that are to be rendered
+        render_instances = []
+        for inst in context:
+            if inst.data["productType"] != "render":
+                # Allow only render instances
+                continue
+
+            if not inst.data.get("publish", True):
+                # Skip inactive instances
+                continue
+
+            if not inst.data.get("farm"):
+                # Only consider instances that are also set to be rendered on
+                # farm
+                continue
+
+            render_instances.append(inst)
+
+        if not render_instances:
+            raise RuntimeError("No instances found for Deadline submission")
+
+        instance.data["_farmRenderInstances"] = render_instances
```

Comment:
So we store other instances to the instance itself?

Reply:
Yes - just so that in this same plug-in we can access the value in one of the other overridable methods. In particular in `get_job_info` that we override from the abstract base class.